infracheck/checks/domain-expiration

```diff
@@ -47,43 +56,77 @@ class DomainCheck:
         if not self._domain or self._domain == '':
             raise Exception('Domain must be specified')
 
-    def _check_with_wait(self):
+    def _check_with_wait(self) -> WhoisResult:
         time.sleep(1)
         max_retries = 10
         retries = 0
 
-        while True:
+        while retries <= max_retries:
             try:
-                query = self.whois.query(self._domain)
+                _stdout_copy = sys.stdout
+
+                # the "whois" library is using print(), sys.stdout needs to be mocked for a moment to not print those
+                # messages
+                try:
```

Comment:
Is it possible to implement "with" clause instead? Not saying it's wrong but using try-finally to handle resources is a bit ugly, I'm just used to Java's (_try-with-resources_) or C#'s (_using_) way :)

Reply:
It would be nice, but I cannot find such possibility there, but I always can be wrong.

Reply:
Changed it a little bit.